DNN Platform/Library/Entities/Portals/PortalStylesController.cs

```diff
@@ -0,0 +1,128 @@
+﻿#region Copyright
+
+// 
+// DotNetNuke® - http://www.dotnetnuke.com
+// Copyright (c) 2002-2018
+// by DotNetNuke Corporation
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
+// of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
+// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+// DEALINGS IN THE SOFTWARE.
+
+#endregion
+
+
+#region usings
+using DotNetNuke.Collections;
+using DotNetNuke.ComponentModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+#endregion
+
+namespace DotNetNuke.Entities.Portals
+{
+    class PortalStylesController : IPortalStylesController
+    {
+        private const string PRIMARY_COLOR = "PrimaryColor";
+        private const string PRIMARY_COLOR_LIGHT = "PrimaryColorLight";
+        private const string PRIMARY_COLOR_DARK = "PrimaryColorDark";
+        private const string PRIMARY_COLOR_CONTRAST = "PrimaryColorContrast";
+        private const string SECONDARY_COLOR = "SecondaryColor";
+        private const string SECONDARY_COLOR_LIGHT = "SecondaryColorLight";
+        private const string SECONDARY_COLOR_DARK = "SecondaryColorDark";
+        private const string SECONDARY_COLOR_CONTRAST = "SecondaryColorContrast";
+        private const string TERTIARY_COLOR = "TertiaryColor";
+        private const string TERTIARY_COLOR_LIGHT = "TertiaryColorLight";
+        private const string TERTIARY_COLOR_DARK = "TertiaryColorDark";
+        private const string TERTIARY_COLOR_CONTRAST = "TertiaryColorContrast";
+        private const string DEFAULT_RADIUS = "DefaultRadius";
+        private const string BUTTON_RADIUS = "ButtonRadius";
+
+        /// <summary>
+        /// Gets an instance of the PortalStylescontroller
+        /// </summary>
+        /// <returns></returns>
+        public static IPortalStylesController Instance()
```

Comment:
This could be a place to use the new DependencyInjection process.  Not sure if we MUST do this here, but it could be a great way to start documenting the "best" way to do things?  Thoughts?

Reply:
Agree, let's use the best practices we recommend...
Now let me find that blog post and try to implement that :)

Reply:
yeah, I tried but not sure how to do this without also converting the whole PortalSettingsController class to the new way... So I say either I need help to figure it out (alway up for some learning) or let's keep this for a separate project. I know only enough about this to break everything :)